packages/manager/src/managers/simulator/SimulatorManager.ts

```diff
@@ -0,0 +1,151 @@
+	t.type({
+		title: t.string,
+		body: t.string,
+	}),
+	t.partial({
+		description: t.string,
+		validate: functionCodec,
+	}),
+]);
+
+const SliceSimulatorSetupStepValidationMessageCodec = t.type({
+	title: t.string,
+	message: t.string,
+});
+type SliceSimulatorSetupStepValidationMessageCodec = t.TypeOf<
+	typeof SliceSimulatorSetupStepValidationMessageCodec
+>;
+
+export type SimulatorManagerReadSliceSimulatorSetupStep = {
+	title: string;
+	description?: string;
+	body: string;
+	/**
+	 * Determines if the step is completed.
+	 *
+	 * This proeprty is `undefined` if the project's adapter does not provide a
+	 * validation function for the step; we cannot know if the step is complete
+	 * without a validator.
+	 */
+	isComplete: boolean | undefined;
+	validationMessages: SliceSimulatorSetupStepValidationMessageCodec[];
+};
+
+export type SimulatorManagerReadSliceSimulatorSetupStepsReturnType = {
+	steps: SimulatorManagerReadSliceSimulatorSetupStep[];
+	errors: (DecodeError | HookError)[];
+};
+
+export class SimulatorManager extends BaseManager {
+	async getLocalSliceSimulatorURL(): Promise<string | undefined> {
+		const sliceMachineConfig = await this.project.getSliceMachineConfig();
+
+		return sliceMachineConfig.localSliceSimulatorURL;
+	}
+
+	/**
+	 * @throws {@link UnexpectedDataError} Thrown if the project is not configured
+	 *   with a Slice Simulator URL.
+	 */
+	async checkIsLocalSliceSimulatorURLAccessible(): Promise<boolean> {
+		const localSliceSimulatorURL = await this.getLocalSliceSimulatorURL();
+
+		if (!localSliceSimulatorURL) {
+			throw new UnexpectedDataError(
+				"The project has not been configured with a Slice Simulator URL. Add a `localSliceSimulatorURL` property to your project's configuration to fix this error.",
+			);
+		}
+
+		const res = await fetch(localSliceSimulatorURL);
+
+		return res.ok;
+	}
+
+	async readSliceSimulatorSetupSteps(): Promise<SimulatorManagerReadSliceSimulatorSetupStepsReturnType> {
+		assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+		const hookResult = await this.sliceMachinePluginRunner.callHook(
+			"slice-simulator:setup:read",
+			undefined,
+		);
+		const { data, errors } = decodeHookResult(
+			t.array(sliceSimulatorSetupStepCodec),
+			hookResult,
+		);
+
+		const steps = await Promise.all(
+			data[0].map(async (step) => {
+				const bodyHTML = await markdownToHTML(step.body);
+
+				const res: SimulatorManagerReadSliceSimulatorSetupStep = {
+					title: step.title,
+					description: step.description,
+					body: bodyHTML,
+					isComplete: undefined,
+					validationMessages: [],
+				};
+
+				if (step.validate) {
+					const validationResult = await step.validate();
+					const { value: validationMessages, error } = decode(
+						t.array(SliceSimulatorSetupStepValidationMessageCodec),
+						validationResult == null ? [] : castArray(validationResult),
+					);
+
+					if (error) {
+						// TODO: We may want to do
```

Comment:
These errors are not user facing, so I think it's safe to ignore them, or at least not return to SM UI.

If that branch is reached, the step is treated as if it had no validation (i.e. `isComplete: undefined`). In the UI, we don't display it as being completed (`isComplete: true`) or incomplete (`isComplete: false`) since we don't know the result.

Plugin authors should catch this error during testing or during compilation if they are written in TypeScript.

What do you think? Should we ignore these errors?